src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -28,15 +27,9 @@ import org.gradle.api.artifacts.repositories.UrlArtifactRepository
 import org.gradle.api.provider.Provider
 import org.gradle.api.publish.PublishingExtension
 import org.gradle.api.publish.plugins.PublishingPlugin
+import org.gradle.api.reflect.TypeOf.typeOf
 import org.gradle.api.tasks.TaskContainer
 import org.gradle.api.tasks.TaskProvider
-import org.gradle.kotlin.dsl.create
-import org.gradle.kotlin.dsl.invoke
-import org.gradle.kotlin.dsl.named
-import org.gradle.kotlin.dsl.register
-import org.gradle.kotlin.dsl.registerIfAbsent
-import org.gradle.kotlin.dsl.typeOf
-import org.gradle.kotlin.dsl.withType
```

Comment:
We can keep these by adding a dependency on kotlin-dsl (not the plugin). It's safe because they're mostly inline afaik and the artifact doesn't change much.

Reply:
That's an option but IMHO it's safer and potentially less error prone to remove it from the plugin entirely. The syntactical improvements that kotlin-dsl provides are nice, but they're really intended for use by build authors so the Kotlin DSL can be aligned as closely as possible to the older Groovy DSL. I think it's better to remove it and not worry about it. But it's a question of convenience vs compatibility and I'll let a maintainer weigh in.

Reply:
@3flex so you're saying that kotlin-dsl was meant for buildSrc only?

@szpak thoughts on extra dep to keep these, or accept the Java-compatible syntax and potentially platform types in some places?

Reply:
I still write my own scripts in Groovy, so I'm not up to date with the Kotlin stuff :-/.

The syntax is definitely "less pretty". What is the official recommendation of the Gradle team for the plugins? Is kotlin-dsl is really for "buildSrc", not for the custom plugins? Does the Gradle team have a plan to avoid that kind of "compatibility" issues with the next Kotlin versions (such as incoming 2.0)?

Reply:
> What is the official recommendation of the Gradle team for the plugins?

I don't think there are any explicit recommendations regarding use of the Kotlin DSL in published third party plugins.

> Is kotlin-dsl is really for "buildSrc", not for the custom plugins?

It's there to support precompiled script plugins, not binary plugins. But the suggestion to apply `gradleKotlinDsl()` as a dependency instead should avoid the version incompatibility in the plugin while allowing use of the Kotlin extensions, as long as everything is backwards compatible to the oldest supported Gradle version. I'll try that.

> Does the Gradle team have a plan to avoid that kind of "compatibility" issues with the next Kotlin versions (such as incoming 2.0)?

No idea what's on the roadmap for Kotlin DSL.

Reply:
@3flex Are you on the Gradle Slack (or their forum)? Maybe it is worth to ask a question about the recommendation in that case? Maybe they have any idea to push it forward (and maybe we would need to switch again to something new soon)?

Reply:
Sorry but I'll have to pass on that - my only motivation was to unblock updating Gradle in this project and this PR is a way to do it without breaking anything at the cost of some less pleasant syntax in some places. Removing `kotlin-dsl` completely will definitely not make compatibility worse. Leaving it in might (or might not).

If you'd like to take a different approach I'm happy to make updates, just let me know the direction you'd like to go in, or we can close this PR if you prefer.

Reply:
I think we should go ahead with the PR and make a 2.0 release, and then later on when we drop Gradle 6.x (requiring old Kotlin), we can maybe reintroduce this.